metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DepublishRecordIdDao.java

```diff
@@ -131,6 +131,37 @@ public int createRecordIdsToBeDepublished(String datasetId, Set<String> candidat
     return recordIdsToAdd.size();
   }
 
+  /**
+   * Deletes a list of record ids from the database. Only record ids that are in a {@link
+   * DepublicationStatus#PENDING_DEPUBLICATION} state will be removed.
+   *
+   * @param datasetId The dataset to which the depublish record ids belong.
+   * @param recordIds The depublish record ids to be removed
+   * @throws BadContentException In case adding the records would violate the maximum number of
+   * depublished records that each dataset can have.
+   * @return The number or record ids that were removed.
+   */
+  public Integer deletePendingRecordIds(String datasetId, Set<String> recordIds)
+      throws BadContentException {
+
+    // Check list size: if this is too large we can throw exception regardless of what's in the database.
+    if (recordIds.size() > maxDepublishRecordIdsPerDataset) {
+      throw new BadContentException(
+          "Can't remove these records: this would violate the maximum number of records per dataset.");
+    }
+
+    final Query<DepublishRecordId> query = morphiaDatastoreProvider.getDatastore()
+        .createQuery(DepublishRecordId.class);
+    query.field(DepublishRecordId.DATASET_ID_FIELD).equal(datasetId);
+    query.field(DepublishRecordId.RECORD_ID_FIELD).in(recordIds);
+    query.field(DepublishRecordId.DEPUBLICATION_STATUS_FIELD)
+        .equal(DepublicationStatus.PENDING_DEPUBLICATION);
```

Comment:
Should we instead not throw an error if any record is found to be undeletable because it has the wrong status? What do you think? Should we silently skip it?

Reply:
I think we should silently skip it yes.
The default behavior of a delete operation is to be silent in general.
And also we expect the incoming list to be the correct one, since we are in control of it.